### PR TITLE
Implement paint worklet properties

### LINKED ISSUE #16839: Implement CSS Houdini Paint Worklets
Draft spec: https://drafts.css-houdini.org/css-paint-api/
Houdini CSS issue for Servo: #13557.
Houdini Worklets issue for Servo: #16206.

### PR DESCRIPTION
**Pull request by asajeffrey**
This is the final PR to get basic paint worklet support. It adds support for paint worklet properties (https://drafts.css-houdini.org/css-paint-api/#paint-definition-input-properties). When a paint worklet is registered, it specifies a list of CSS properties, and is provided with their computed values when it is invoked.

This is a dependent PR:
* "Implemented paint worklets invoking worklet scripts" is #17239.
* "Implemented paint worklets rendering contexts" is #17326.

There should be tests added for this, hopefully the existing wpt houdini tests.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #16839
- [x] There are tests for these changes

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17364)

### CODE REVIEW

**Comment by asajeffrey** [comment]
cc @metajack @tschneidereit @jdm.

**Comment by asajeffrey** [comment]
The simplest Google demo now renders

**Comment by asajeffrey** [comment]
out

**Comment by asajeffrey** [comment]
Rebased. This PRs dependencies have all landed, so it's ready for review.

**Comment by asajeffrey** [comment]
The wpt css-paint-api tests are now in `/_mozilla/mozilla/css-paint-api`, so we have tests!

r? @jdm

File: components/script/dom/styepropertymapreadonly.rs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (jdm) [inline components/script/dom/styepropertymapreadonly.rs]: Typoed filename that doesn't belong here.

File: components/layout/display_list_builder.rs [hunk 2]
```diff
@@ -1165,17 +1165,24 @@ impl FragmentDisplayListBuilding for Fragment {
...
-        // If the script thread has not added any paint worklet modules, there is nothing to do!
-        let executor = match state.layout_context.paint_worklet_executor {
-            Some(ref executor) => executor,
-            None => return debug!("Worklet {} called before any paint modules are added.", name),
+        // Get the painter, and the computed values for its properties.
+        let (properties, painter) = match state.layout_context.registered_painters.read().get(&name) {
+            Some(registered_painter) => (
+                registered_painter.properties
+                    .iter()
+                    .filter_map(|(name, id)| id.as_shorthand().err().map(|id| (name, id)))
```
Comment (jdm) [inline components/layout/display_list_builder.rs]: Didn't we already filter them when registering?

**Review by jdm — CHANGES_REQUESTED** [review]

File: components/script/dom/styepropertymapreadonly.rs (hunk 1 above)
Comment (asajeffrey) [inline components/script/dom/styepropertymapreadonly.rs]: Oops.

File: components/layout/display_list_builder.rs (hunk 2 above)
Comment (asajeffrey) [inline components/layout/display_list_builder.rs]: Yes, but very annoyingly shorthand property ids have a lifetime, so we can't store them in the properties map.

**Review by asajeffrey — COMMENTED** [review]
Oops, removed that filetypo!

**Comment by jdm** [comment]
Looks good to merge after squashing, in that case.

**Review by jdm — APPROVED** [review]

**Comment by asajeffrey** [comment]
Squashed. @bors-servo: r=jdm

**Comment by jdm** [comment]
```
  ▶ Unexpected subtest result in /_mozilla/mozilla/interfaces.html:
  │ FAIL [expected PASS] Interfaces exposed on the window
  │   → assert_true: If this is failing: DANGER, are you sure you want to expose the new interface CSSStyleValue to all webpages as a property on the global? Do not make a change to this file without review from jdm or Ms2ger for that specific change! expected true got false
  │ 
  │ test_interfaces/<@http://web-platform.test:8000/_mozilla/mozilla/interfaces.js:73:7
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1406:20
  │ test@http://web-platform.test:8000/resources/testharness.js:497:9
  │ test_interfaces@http://web-platform.test:8000/_mozilla/mozilla/interfaces.js:2:3
  └ @http://web-platform.test:8000/_mozilla/mozilla/interfaces.html:13:1
```

**Comment by asajeffrey** [comment]
Made the new interfaces just exposed to Worklet, not Window. @bors-servo r=jdm

**Pull request merged**